### PR TITLE
feat: add ability to attach access tags to resources

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
Add access tags to the postgresql resource created by the module and update tests

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content
Add ability to attach access tags to resources created by the module, consumers who want to use the access tags will need additional permissions for Tagging service at the Administrator level

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +34,12 @@ You need the following permissions to run this module.
 - Account Management
     - **Databases for PostgreSQL** service
         - `Editor` role access
+
+Optionally, you need the following permissions to attach Access Management tags to resources in this module.
+
+- IAM Services
+    - **Tagging** service
+        - `Administrator` platform access
```
Comment (SirSpidey) [inline README.md:42]:
```suggestion
To attach access management tags to resources in this module, you need the following permissions.

- IAM Services
    - **Tagging** service
        - `Administrator` platform access
```

If it's optional, you don't "need" it.

File: examples/complete/variables.tf [hunk 2]
```diff
@@ -28,6 +28,12 @@ variable "resource_tags" {
   default     = []
 }
 
+variable "access_tags" {
+  type        = list(string)
+  description = "Optional list of access management tags to be added to created resources"
```
Comment (SirSpidey) [inline examples/complete/variables.tf:33]:
```suggestion
  description = "Optional list of access management tags to add to resources that are created"
```

File: examples/default/variables.tf [hunk 3]
```diff
@@ -28,6 +28,12 @@ variable "resource_tags" {
   default     = []
 }
 
+variable "access_tags" {
+  type        = list(string)
+  description = "Optional list of access management tags to be added to created resources"
```
Comment (SirSpidey) [inline examples/default/variables.tf:33]:
```suggestion
  description = "Optional list of access management tags to add to resources that are created"
```

File: examples/fscloud/variables.tf [hunk 4]
```diff
@@ -28,6 +28,12 @@ variable "resource_tags" {
   default     = []
 }
 
+variable "access_tags" {
+  type        = list(string)
+  description = "Optional list of access management tags to be added to created resources"
```
Comment (SirSpidey) [inline examples/fscloud/variables.tf:33]:
```suggestion
  description = "Optional list of access management tags to add to resources that are created"
```

File: examples/pitr/variables.tf [hunk 5]
```diff
@@ -28,6 +28,12 @@ variable "resource_tags" {
   default     = []
 }
 
+variable "access_tags" {
+  type        = list(string)
+  description = "Optional list of access management tags to be added to created resources"
```
Comment (SirSpidey) [inline examples/pitr/variables.tf:33]:
```suggestion
  description = "Optional list of access management tags to add to resources that are created"
```

File: examples/replica/variables.tf [hunk 6]
```diff
@@ -28,6 +28,12 @@ variable "resource_tags" {
   default     = []
 }
 
+variable "access_tags" {
+  type        = list(string)
+  description = "Optional list of access management tags to be added to created resources"
```
Comment (SirSpidey) [inline examples/replica/variables.tf:33]:
```suggestion
  description = "Optional list of access management tags to add to resources that are created"
```

File: module-metadata.json [hunk 7]
```diff
@@ -1,14 +1,35 @@
 {
   "path": ".",
   "variables": {
+    "access_tags": {
+      "name": "access_tags",
+      "type": "list(string)",
+      "description": "A list of access tags to apply to the resources created by the module, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial for more details",
```
Comment (SirSpidey) [inline module-metadata.json:7]:
```suggestion
      "description": "A list of access tags to apply to the resources created by the module. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial.",
```

File: profiles/fscloud/variables.tf [hunk 8]
```diff
@@ -62,6 +62,12 @@ variable "resource_tags" {
   default     = []
 }
 
+variable "access_tags" {
+  type        = list(string)
+  description = "Optional list of access management tags to be added to created resources"
```
Comment (SirSpidey) [inline profiles/fscloud/variables.tf:67]:
```suggestion
  description = "Optional list of access management tags to add to resources that are created"
```

File: variables.tf [hunk 9]
```diff
@@ -152,6 +152,19 @@ variable "resource_tags" {
   default     = []
 }
 
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the resources created by the module, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial for more details"
```
Comment (SirSpidey) [inline variables.tf:157]:
```suggestion
  description = "A list of access tags to apply to the resources created by the module. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
```

**Review by SirSpidey — CHANGES_REQUESTED** [review]
I think I caught all the user content about tags.

**Comment by MatthewLemmond** [comment]
The PR test is failing with:
```Error updating database cpu auto_scaling group: Request failed with status code: 422, ServerErrorResponse: {"errors":{"cpu.rate.limit_count_per_member":["cannot autoscale higher than regular scaling limit"]}}```

From what I can see none of the scaling variables were affected by my PR (diffs only show addition of access tag variables), is there a change that was recently merged that I need to account for/add a variable for in order to pass the test?

**Comment by MatthewLemmond** [comment]
Looks like that resolved the issue, thanks @ocofaigh @shemau!

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Can you update all the variable description. Access tags only get applied to the postgres instance as far as I can see, so perhaps update:

`to apply to the resources created by the module` -> `to apply to the PostgreSQL instance created by the module`

**Review by SirSpidey — DISMISSED** [review]
Content updates look good. Thanks.

File: profiles/fscloud/variables.tf [hunk 10]
```diff
@@ -70,6 +70,12 @@ variable "resource_tags" {
   default     = []
 }
 
+variable "access_tags" {
+  type        = list(string)
+  description = "Optional list of access management tags to add to resources that are created"
```
Comment (ocofaigh) [inline profiles/fscloud/variables.tf:75]: Should match the variable description in the root level variables.tf

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**